corehq/apps/callcenter/fixturegenerators.py

```diff
@@ -13,17 +15,26 @@ def should_sync(domain, last_sync, utcnow=None):
     if not last_sync or not last_sync.date:
         return True
 
+    # utcnow only used in tests to mock other times
+    utcnow = utcnow or datetime.utcnow()
+
     try:
         timezone = domain.get_default_timezone()
     except pytz.UnknownTimeZoneError:
         timezone = utc
 
-    # check if user has already synced today (in local timezone). Indicators only change daily.
-    last_sync_utc = last_sync.date if last_sync.date.tzinfo else utc.localize(last_sync.date)
-    last_sync_local = timezone.normalize(last_sync_utc.astimezone(timezone))
+    _assert = soft_assert(to=['droberts' + '@' + 'dimagi.com'])
+
+    last_sync_utc = last_sync.date
+
+    if not _assert(last_sync_utc.tzinfo is None,
+                   'last_sync.date should be an offset-naive dt'):
+        last_sync_utc = UserTime(last_sync_utc).server_time().done()
```

Comment:
Yeah, it's assumed to be server time—thus the soft-assert that tzinfo is None—but if there *is* a timezone, then this emulates `last_sync_local = timezone.normalize(last_sync_utc.astimezone(timezone))` above. In other words, (1) I think this will never happen, (2) there's no behavior change before and after this commit if it does happen. In the line you linked to specifically, this line of code won't be called. If `last_sync.date` comes in with a timezone, it'll be converted to UTC (respecting the tzinfo) and then have the timezone removed—that's both what we want and what `UserTime(x).server_time().done()` does.